elastic-search/elastic-search.yaml

```diff
@@ -0,0 +1,543 @@
+                  - "24224:24224/udp"
+                network_mode: host
+            EOF
+
+            # Create the Dockerfile
+            mkdir -p /home/ubuntu/fluentd
+            cat <<'EOF' > /home/ubuntu/fluentd/Dockerfile
+            FROM fluent/fluentd:v1.16-1
+
+            # Use root account to use apk
+            USER root
+
+            # below RUN includes plugin as examples elasticsearch is not required
+            # you may customize including plugins as you wish
+            RUN apk add --no-cache --update --virtual .build-deps \
+                sudo build-base ruby-dev \
+            && sudo gem install fluent-plugin-elasticsearch \
+            && sudo gem sources --clear-all \
+            && apk del .build-deps \
+            && rm -rf /tmp/* /var/tmp/* /usr/lib/ruby/gems/*/cache/*.gem
+            USER fluent
+            EOF
+
+            # Create the Fluentd config file
+            mkdir -p /home/ubuntu/fluentd/conf
+            cat <<'EOF' > /home/ubuntu/fluentd/conf/fluent.conf
+            <source>
+              @type http
+              port 24224
+              bind 0.0.0.0
+            </source>
+
+            <match *.**>
+              @type copy
+
+              <store>
+                @type elasticsearch
+                host ${ESALB.DNSName}
+                port 9200
+                logstash_format true
+                logstash_prefix fluentd
+                logstash_dateformat %Y%m%d
+                include_tag_key true
+                type_name access_log
+                tag_key @log_name
+                flush_interval 1s
+              </store>
+
+              <store>
+                @type stdout
+              </store>
+            </match>
+            EOF
+
+            # Change ownership of the files to ubuntu user
+            sudo chown -R ubuntu:ubuntu /home/ubuntu/docker-compose.yml /home/ubuntu/fluentd
+            # Start Fluentd using Docker Compose
+            sudo -u ubuntu /home/ubuntu/.docker/cli-plugins/docker-compose -f /home/ubuntu/docker-compose.yml up -d
+
+  EC2InstanceForESBuild:
+    Type: 'AWS::EC2::Instance'
+    Properties:
+      InstanceType: t2.micro
+      SubnetId: !Ref "AWS::NoValue"
+      SecurityGroupIds:
+        - !Ref ElasticSecurityGroup
+      KeyName: !Ref KeyName
+      ImageId: ami-04b70fa74e45c3917 # Ubuntu 24.04 LTS
+      Tags:
+        - Key: Name
+          Value: es-cluster-builder
+      UserData:
+        Fn::Base64:
+          !Sub |
+            #!/bin/bash
+            apt-get update -y > /var/log/user-data.log 2>&1
+            apt-get install -y curl unzip > /var/log/user-data.log 2>&1
+            cd /home/ubuntu
+            curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip" > /var/log/user-data.log 2>&1
+            unzip awscliv2.zip > /var/log/user-data.log 2>&1
+            ./aws/install > /var/log/user-data.log 2>&1
+            aws configure set aws_access_key_id ${AccessKeyId}
```

Comment:
TODO: Replace the usage of AccessKeyId and SecretAccessKey with IAM roles for enhanced security and best practices.